src/lm/arpa-file-parser.h

```diff
@@ -86,9 +91,35 @@ class ArpaFileParser {
   void Read(std::istream &is, bool binary);
 
   /// Parser options.
-  const ArpaParseOptions& Options() const { return options_; }
+  const ArpaParseOptions &Options() const { return options_; }
 
  protected:
+  /// A status value returned from ConsumeNGram() overload.
+  class Status {
+   public:
+    /// Return the warninig status to print a warninig and continue.
+    static Status Warning(const std::string& message) {
+      return Status(kWarninig, message); }
```

Comment:
Well, I did not do that for the love of classes. The diagnostic was really poor: I could print a line number in ARPA file, but if that comes out of pipeline, good luck finding it. So I need to communicate back from ConsumeNGram what it decided to do with it: accept, accept with a warning or die fatally. Formatting is done in the common class, and is actually useful:for locating the problem:
```
line 21 [-1.5       a β -3.2] skipped: word 'β' not in symbol table.
line 24 [-0.03333        <s> b b b] skipped: no parent (n-1)-gram exists
```
Rate limiting *by itself* is not a problem, the counter is a private member of the ARPA parser class. The question is *what* to count. I would really hate to increment a global variable defined in one file from two others every time I print KALDI_WARN. It's probably even worse.